src/app/ReadClient.cpp

```diff
@@ -884,6 +884,7 @@ void ReadClient::OnLivenessTimeoutCallback(System::Layer * apSystemLayer, void *
                 }
 
                 session->MarkAsDefunct();
+                session->DispatchSessionEvent(&SessionDelegate::OnSessionHang);
```

Comment:
It's not clear to me that this makes sense.  The only session we have evidence of a hang on is the one we're actually using, right?

In general, the semantics of OnSessionHang are ... not very clearly defined.

Reply:
This change will be useful for detect moment when device is connected without DeviceCommissioner polling like that

```
chip::SessionDelegate::NewSessionHandlingPolicy GetNewSessionHandlingPolicy() override {
    // handle device connected

    return NewSessionHandlingPolicy::kShiftToNewSession;
}

void OnSessionHang() override {
    // handle device disconnected
}
````

Reply:
My point is that the session being marked defunct here may not have any issues with it.  We are marking a _bunch_ of session defunct, and the only one we have evidence for being bad is the one that actually timed out....

Reply:
Are you suggesting to use ReadClient::Callback::OnResubscriptionNeeded/OnSubscriptionEstablished for subscriptions instead of SessionDelegate?

Reply:
@rus084 I am not suggesting anything so far.  I am pointing out that what this PR does is very odd.

If you are looking for a suggestion: what specific problem are you trying to solve?

Reply:
i'm trying to detect when device connects or disconnects from network
Seems use OnResubscriptionNeeded/OnSubscriptionEstablished with subscriptions is good way